hudi-flink-datasource/hudi-flink/src/main/java/org/apache/hudi/util/ClusteringUtil.java

```diff
@@ -77,4 +77,21 @@ public static void rollbackClustering(HoodieFlinkTable<?> table, HoodieFlinkWrit
       table.getMetaClient().reloadActiveTimeline();
     });
   }
+
+  /**
+   * Force rolls back the inflight clustering instant, for handling failure case.
+   *
+   * @param table The hoodie table
+   * @param writeClient The write client
+   * @param instantTime The instant time
+   */
+  public static void rollbackClustering(HoodieFlinkTable<?> table, HoodieFlinkWriteClient writeClient, String instantTime) {
+    Option<HoodieInstant> inflightInstant = Option.fromJavaOptional(ClusteringUtils.getPendingClusteringInstantTimes(table.getMetaClient())
+        .stream().filter(instant -> instant.equals(instantTime)).findFirst());
+    if (inflightInstant.isPresent()) {
```

Comment:
instant.getTimestamp().equals(instantTime)

Reply:
How about we change to this code:

```java
    HoodieInstant inflightInstant = HoodieTimeline.getReplaceCommitInflightInstant(instantTime);
    if (table.getMetaClient().reloadActiveTimeline().filterPendingReplaceTimeline().containsInstant(inflightInstant)) {
      LOG.warn("Rollback failed clustering instant: [" + instantTime + "]");
      table.rollbackInflightClustering(inflightInstant,
          commitToRollback -> writeClient.getPendingRollbackInfo(table.getMetaClient(), commitToRollback, false));
    }
```

There is no need to deserialize the clustering plan because we definitely know it is a clustering instant.

Reply:
@danny0405, this is indeed definitely know it is a clustering instant.